File: app/src/main/java/com/refrii/client/Box.java
package com.refrii.client;

import java.io.Serializable;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;

public class Box implements Serializable {
    private int id;
    private String name;
    private String notice;
    private String imageUrl;
    private boolean isInvited;
    private Date updatedAt;
    private Date createdAt;
    private List<Food> foods;
    private List<User> invitedUsers;
    private User owner;

    public Box(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getNotice() {
        return notice;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public boolean isInvited() {
        return isInvited;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public List<Food> getFoods() {
        return foods;
    }

    public List<User> getInvitedUsers() {
        return invitedUsers;
    }

    public User getOwner() {
        return owner;
    }

    public boolean equals(Object object) {
        Box box = (Box) object;
        return box.id == id;
    }
}

File: app/src/main/java/com/refrii/client/Box.kt
package com.refrii.client

import android.content.Context
import retrofit2.Call
import retrofit2.Response
import rx.Observable
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers
import java.io.Serializable
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Date

class Box(val id: Int) : Serializable {
    var name: String? = null
    var notice: String? = null
    val imageUrl: String? = null
    val isInvited: Boolean = false
    val updatedAt: Date? = null
    val createdAt: Date? = null
    val foods: MutableList<Food>? = null
    val invitedUsers: List<User>? = null
    val owner: User? = null

    override fun equals(obj: Any?): Boolean {
        obj ?: return false

        val box = obj as Box
        return box.id == id
    }

    companion object {
        fun getAll(context: Context): Observable<List<Box>> {
            return RetrofitFactory.getClient(BoxService::class.java, context).getBoxes()
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
        }
    }
}
